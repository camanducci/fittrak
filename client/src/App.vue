<template>
<div id="app" @click="away">
  <div class="container is-fluid">
    <Topbar></Topbar>
    <Sidebar></Sidebar>

    <div class="columns main">
      <div class="column">
        <router-view></router-view>
      </div>
    </div>
  </div> <!-- end container -->
</div>
</template>

<style lang="scss">
@import '../node_modules/bulma/css/bulma';
@import '../node_modules/font-awesome/scss/font-awesome';

div.main{
  margin-top: 3rem;
}

div.main h2 {
  margin: 0 1rem;
  text-align: center;
  font-size: 2rem;
}

form {
  padding: 1rem;
  margin: 1rem;
  background: white;
}

form label {
  display: block;
}

form input[type=submit] {
  display: block;
}
</style>


<script>
import Topbar from './components/Topbar'
import Sidebar from './components/Sidebar'

export default {
  methods: {
    away (e) {
      let clsName = e.target.className
      let showing = this.$store.getters.showSidebar

      // Close sidebar if clicked outside
      if (showing) {
        if (clsName !== 'sidebar active' && clsName !== 'fa fa-bars toggle-icon') {
          this.$store.dispatch('toggleSidebar')
        }
      }
    }
  },
  components: {
    Topbar,
    Sidebar
  }
}
</script>
