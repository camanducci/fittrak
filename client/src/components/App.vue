<template>
<v-app>
  <Sidebar v-if="authed" v-bind:drawer="drawer"></Sidebar>

  <v-toolbar app clipped-left fixed color="cyan" dark flat>
    <v-toolbar-side-icon 
      app 
      v-if="authed" 
      @click.stop="drawer = !drawer"
    >
    </v-toolbar-side-icon> 

    <v-toolbar-title 
      color="white"
    >FitTrack</v-toolbar-title>

     <v-spacer></v-spacer>

    <v-toolbar-items v-if="!authed">
      <v-btn flat @click.prevent="$router.push('login')">Login</v-btn>
      <v-btn flat @click.prevent="$router.push('register')">Register</v-btn>
    </v-toolbar-items>
    <v-flex align-center v-else class="text-xs-right">
      Heya, <strong>{{username}}</strong>
    </v-flex>

  </v-toolbar>

  <main>
    <v-content>
      <v-container fluid fill-height>
        <v-layout justify-center align-center>
          <router-view></router-view>
        </v-layout>
      </v-container>
    </v-content>
  </main>
  <v-footer app class="pa-2">
    <v-layout justify-center>
      Made with <span style="color:red;">&#10084;</span> by <a href="https://github.com/tsoporan/fittrack">tsoporan</a>
    </v-layout>
  </v-footer>
</v-app>
</template>

<script>
import Sidebar from './Sidebar'

import auth from '../auth'

export default {
  data: () => ({
    drawer: false
  }),
  computed: {
    authed () {
      return auth.user.authed
    },
    username () {
      return auth.user.username
    }
  },
  components: {
    Sidebar
  }
}
</script>
