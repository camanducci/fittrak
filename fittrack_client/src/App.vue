<template>
<div id="app">
    <div class="container is-fluid">
        <Header></Header>
        <Sidebar></Sidebar>

        <div class="columns main">
            <div class="column"> 
            </div>
        </div>

        <Footer></Footer>
    </div> <!-- end container -->
</div>
</template>


<style lang="scss">
@import '../node_modules/bulma/css/bulma';
@import '../node_modules/font-awesome/scss/font-awesome';

div.main{
  margin-top: 3rem;
}

div.main h2 {
  margin: 0 1rem;
  text-align: center;
  font-size: 2rem;
}

form {
  padding: 1rem;
  margin: 1rem;
  background: white;
} 

form label {
  display: block;
} 

form input[type=submit] {
  display: block;
}
</style>


<script>
import Header from './components/Header'
import Sidebar from './components/Sidebar'
import Footer from './components/Footer'

export default {
  components: {
    Header,
    Sidebar,
    Footer
  }
}
</script>
