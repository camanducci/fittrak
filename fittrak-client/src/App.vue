<template>
<v-app id="fittrak">
  <v-navigation-drawer
    v-model="drawer"
    fixed
    app
    dark
  >
    <SidebarNavigationItems  />
  </v-navigation-drawer>

  <v-toolbar color="primary" dark fixed app>
    <v-toolbar-side-icon @click.stop="drawer = !drawer"></v-toolbar-side-icon>
    <v-toolbar-title class="logo">FitTrak</v-toolbar-title>
    <v-spacer></v-spacer>
    <v-toolbar-items>
      <p class="viewer">Heya, <strong>{{ viewer }}</strong></p>
    </v-toolbar-items>
  </v-toolbar>

  <v-content>
    <v-container fluid fill-height>
      <v-layout row wrap>
        <v-flex xs12>
          <router-view />
        </v-flex>
      </v-layout>
    </v-container>
  </v-content>
</v-app>
</template>

<script>
import VIEWER from "@/graphql/queries/viewer.graphql";

import SidebarNavigationItems from "@/components/sidebar/SidebarNavigationItems";

export default {
  name: "App",

  data: () => ({
    drawer: null,
    viewer: "Stranger"
  }),

  props: {
    source: String
  },

  components: {
    SidebarNavigationItems
  },

  apollo: {
    viewer: {
      query: VIEWER,
      update: data => data.viewer.username
    }
  }
};
</script>

<style>
.logo {
  font-family: "Kaushan Script", "Roboto", "Arial";
  font-size: 32px;
}
.viewer {
  margin-top: 20px;
}
</style>
