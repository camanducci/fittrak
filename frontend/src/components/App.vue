<template>
<v-app>
  <Sidebar v-if="authed" v-bind:drawer="drawer"></Sidebar>

  <v-toolbar app clipped-left fixed color="cyan" dark flat>
    <v-toolbar-side-icon 
      app 
      v-if="authed" 
      @click.stop="drawer = !drawer"
    >
    </v-toolbar-side-icon> 

    <v-toolbar-title 
      color="white"
      style="font-family: Kaushan Script; font-size: 24px"
    >
    FitTrack
    </v-toolbar-title>

    <v-spacer></v-spacer>

    <v-toolbar-items v-if="!authed">
      <v-btn flat @click.prevent="$router.push('/signin')">Sign In</v-btn>
      <v-btn flat @click.prevent="$router.push('/register')">Register</v-btn>
    </v-toolbar-items>
    <v-flex align-center v-else class="text-xs-right">
      Heya, <strong>{{ username }}</strong>
    </v-flex>

  </v-toolbar>

  <main>
    <v-content>
      <template v-if="authed && !initialFetch">
        <v-container fluid fill-height>
          <Loader></Loader>
        </v-container>
      </template>
      <template v-else>
      <v-container fluid fill-height>
        <router-view></router-view>
      </v-container>
      </template>
    </v-content>
  </main>

  <Footer></Footer>
  </v-app>
</template>

<script>
import { mapGetters, mapState } from 'vuex'

import Sidebar from './Sidebar'
import Footer from './Footer'
import Loader from './Loader'

export default {
  data: () => ({
    drawer: false
  }),
  computed: {
    ...mapState({
      authed: state => state.user.authed,
      username: state => state.user.username
    }),
    ...mapGetters(['initialFetch'])
  },
  components: {
    Sidebar,
    Footer,
    Loader
  }
}
</script>
