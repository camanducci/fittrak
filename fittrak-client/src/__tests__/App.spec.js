import { shallowMount } from "@vue/test-utils";

import App from "../App";

describe("App.vue", () => {
  test("App renders", () => {
    const app = shallowMount(App);
    expect(app.html()).toMatchSnapshot();
  });
});
