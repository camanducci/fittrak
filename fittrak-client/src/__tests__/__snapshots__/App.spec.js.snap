// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue App renders 1`] = `
<div id="app">
  <nav>
    <router-link to="/">Home</router-link> |
    <router-link to="/history">History</router-link> |
    <router-link to="/settings">Settings</router-link> |
    <a href="/accounts/logout/">Sign out</a>
  </nav>
  <router-view></router-view>
</div>
`;
